rmgpy/molecule/molecule.py

```diff
@@ -1552,11 +1553,9 @@ def updateLonePairs(self):
                         order = order + 3
                     if bond12.isBenzene():
                         order = order + 1.5
-
-                if atom1.isSilicon() or atom1.isCarbon():
-                    atom1.lonePairs = (4 - atom1.radicalElectrons - int(order)) / 2
-                else:     
-                    atom1.lonePairs = 4 - atom1.radicalElectrons - int(order)
+                atom1.lonePairs = (PeriodicSystem.valence_electrons[atom1.symbol] - atom1.radicalElectrons - int(order)) / 2
+                if atom1.lonePairs % 1 > 0:
```

Comment:
Since the calculation above is with all integer types, won't this condition never be triggered? Should you change the equation to divide by 2.0?